// clas/index.js
function indexOf(arr, item) {
	return Array.prototype.indexOf.call(arr, item);
}

function get(element) {
	return element.classList ? element.classList : element.className.split(/\s+/);
}

function has(element, name) {
	return element.classList ? element.classList.contains(name) : !! ~indexOf(get(element), name);
}

function makeClassName(classes) {
	return classes.join(' ').replace(/^\s+|\s+$/g, '');
}

function add(element, name) {
	if (element.classList) {
		element.classList.add(name);
	} else {
		var classes = get(element);
		var index = indexOf(classes, name);
		if (!~index) {
			classes.push(name);
			element.className = makeClassName(classes);
		}
	}
	return element;
}

function remove(element, name) {
	if (element.classList) {
		element.classList.remove(name);
	} else {
		var classes = get(element);
		var index = indexOf(classes, name);
		if (~index) {
			classes.splice(index, 1);
			element.className = makeClassName(classes);
		}
	}
	return element;
}

function toggle(element, name) {
	if (element.classList) {
		element.classList.toggle(name);
		return element;
	} else {
		return has(element, name) ? remove(element, name) : add(element, name);
	}
}

module.exports = {
	get: get,
	has: has,
	add: add,
	remove: remove,
	toggle: toggle,
};
