// clas/build/clas.js
!function(e){if("object"==typeof exports&&"undefined"!=typeof module)module.exports=e();else if("function"==typeof define&&define.amd)define([],e);else{var f;"undefined"!=typeof window?f=window:"undefined"!=typeof global?f=global:"undefined"!=typeof self&&(f=self),f.clas=e()}}(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
function indexOf(arr, item) {
	return Array.prototype.indexOf.call(arr, item);
}

function get(element) {
	return element.classList ? element.classList : element.className.split(/\s+/);
}

function has(element, name) {
	return element.classList ? element.classList.contains(name) : !! ~indexOf(get(element), name);
}

function makeClassName(classes) {
	return classes.join(' ').replace(/^\s+|\s+$/g, '');
}

function add(element, name) {
	if (element.classList) {
		element.classList.add(name);
	} else {
		var classes = get(element);
		var index = indexOf(classes, name);
		if (!~index) {
			classes.push(name);
			element.className = makeClassName(classes);
		}
	}
	return element;
}

function remove(element, name) {
	if (element.classList) {
		element.classList.remove(name);
	} else {
		var classes = get(element);
		var index = indexOf(classes, name);
		if (~index) {
			classes.splice(index, 1);
			element.className = makeClassName(classes);
		}
	}
	return element;
}

function toggle(element, name) {
	if (element.classList) {
		element.classList.toggle(name);
		return element;
	} else {
		return has(element, name) ? remove(element, name) : add(element, name);
	}
}

module.exports = {
	get: get,
	has: has,
	add: add,
	remove: remove,
	toggle: toggle,
};
},{}]},{},[1])(1)
});
